aiodocker/containers.py

```diff
@@ -56,18 +56,31 @@ def __init__(self, docker):
         return DockerContainer(self.docker, id=data['Id'])
 
     async def run(self, config, *, name=None):
+        """
+        Create and start a container.
 
+        If container.start() will raise an error the exception will contain
+        a `container_id` attribute with the id of the container.
+        """
         try:
             container = await self.create(config, name=name)
-        except DockerError as e:
+        except DockerError as err:
             # image not find, try pull it
-            if e.status == 404 and 'Image' in config:
+            if err.status == 404 and 'Image' in config:
                 await self.docker.pull(config['Image'])
                 container = await self.create(config, name=name)
             else:
-                raise e
+                raise err
+
+        try:
+            await container.start()
+        except DockerError as err:
+            raise DockerError(
```

Comment:
Should we have separate exception here? something like DockerContainerError subclass of DockerError and provider of additional property container_id?

Reply:
I implemented this under `class DockerError(Exception)` for now.

I am ok adding a new subclass, but for now, it will be just used by` docker.containers.run`.